pkg/roll/execute.go

```diff
@@ -49,12 +49,15 @@ func (m *Roll) Start(ctx context.Context, migration *migrations.Migration, cbs .
 				fmt.Errorf("unable to execute start operation: %w", err),
 				errRollback)
 		}
-
-		if _, ok := op.(migrations.RequiresSchemaRefreshOperation); ok {
-			// refresh schema
-			newSchema, err = m.state.ReadSchema(ctx, m.schema)
-			if err != nil {
-				return fmt.Errorf("unable to refresh schema: %w", err)
+		// refresh schema when the op is isolated and requires a refresh (for example raw sql)
+		// we don't want to refresh the schema if the operation is not isolated as it would
+		// override changes made by other operations
+		if refreshOp, ok := op.(migrations.RequiresSchemaRefreshOperation); ok && refreshOp.RequiresSchemaRefresh() {
+			if isolatedOp, ok := op.(migrations.IsolatedOperation); ok && isolatedOp.IsIsolated() {
```

Comment:
Main reason is: that it could be non-isolated & require schema refresh. That's the case for migrations with `onComplete`.

Migrations that are not isolated & require a schema refresh on `Start` phase would surely break others, as it would override the resulting schema from the start ops, by refreshing it afterwards. I tried to capture this in the comment